questions/148/explanation.md

```diff
@@ -1 +1,9 @@
-The issue here is not the missing initializer of the variable `a` - it will implicitly be initialized to `0` here. But the issue is the access to `a` twice without sequencing between the accesses. According to §[intro.execution]¶14, accesses of volatile glvalues are side-effects and according to §[intro.execution]¶17 these two unsequenced side-effects on the same memory location results in undefined behavior.
+The issue here is not the missing initializer of the variable `a` - it will implicitly be initialized to `0` since it has static storage duration. The issue is rather that the two accesses of `a` in the expression `a + a` are not sequenced, which leads to undefined behavior.
+
+According to §[intro.execution]¶7, reading the `volatile` `a` is a side effect:
+> Reading an object designated by a volatile glvalue, modifying an object, calling a library I/O function, or calling a function that does any of those operations are all *side effects*
+
+And these side effecting reads are unsequenced, which results in undefined behavior. §[intro.execution]¶10:
+> Except where noted, evaluations of operands of individual operators and of subexpressions of individual expressions are unsequenced.
+> (...)
+> If a side effect on a memory location is unsequenced relative to either another side effect on the same memory location or a value computation using the value of any object in the same memory location, and they are not potentially concurrent, the behavior is undefined.
```

Comment:
I would explain what "potentially concurrent" means or at least provide the same ref as in the quote - [intro.multithread].

Reply:
Yeah, good point. I didn't want to complicate this explanation by talking about concurrency, and considered just replacing that part with "(...)". But the middle thing I ended up with where I kept it in without explanation was not optimal. I've added a reference and a short explanation now. What do you think?